src/main/java/org/folio/circulation/domain/RequestServiceUtility.java

```diff
@@ -113,26 +118,75 @@ private RequestServiceUtility() { }
     }
   }
 
-  static Result<RequestAndRelatedRecords> refuseWhenUserHasAlreadyRequestedItem(
-    RequestAndRelatedRecords request) {
+  static Result<RequestAndRelatedRecords> refuseWhenAlreadyRequested(
+    RequestAndRelatedRecords requestAndRelatedRecords) {
 
-    Optional<Request> requestOptional = request.getRequestQueue().getRequests().stream()
-      .filter(it -> isTheSameRequester(request, it) && it.isOpen()).findFirst();
+    Request request = requestAndRelatedRecords.getRequest();
 
-    if (requestOptional.isPresent()) {
-      Map<String, String> parameters = new HashMap<>();
-      parameters.put("requesterId", request.getRequest().getUserId());
-      parameters.put(ITEM_ID, request.getRequest().getItemId());
-      parameters.put("requestId", requestOptional.get().getId());
-      String message = "This requester already has an open request for this item";
-      return failedValidation(new ValidationError(message, parameters));
+    Predicate<Request> isAlreadyRequested;
+    Function<Request, Result<RequestAndRelatedRecords>> alreadyRequestedErrorMapper;
+
+    if (isTlrEnabled(request)) {
+      alreadyRequestedErrorMapper = (request.getRequestLevel() == TITLE)
+        ? req -> requestLevelTitleAndTlrEnabledErrorMapper(requestAndRelatedRecords)
+        : req -> requestLevelItemAndTlrEnabledErrorMapper(requestAndRelatedRecords);
+      isAlreadyRequested = req -> isTheSameRequester(requestAndRelatedRecords, req);
     } else {
-      return of(() -> request);
+      alreadyRequestedErrorMapper = req -> requestLevelItemAndTlrDisabledErrorMapper(requestAndRelatedRecords);
+      isAlreadyRequested = req -> requestAndRelatedRecords.getItemId().equals(req.getItemId())
+        && isTheSameRequester(requestAndRelatedRecords, req) && req.isOpen();
     }
+
+    return requestAndRelatedRecords.getRequestQueue().getRequests().stream()
+      .filter(isAlreadyRequested)
+      .findFirst()
+      .map(alreadyRequestedErrorMapper)
+      .orElse(of(() -> requestAndRelatedRecords));
+  }
+
+  private static boolean isTlrEnabled(Request request) {
+    TlrSettingsConfiguration tlrSettingsConfiguration = request.getTlrSettingsConfiguration();
+    return tlrSettingsConfiguration != null
+      && tlrSettingsConfiguration.isTitleLevelRequestsFeatureEnabled();
   }
 
   static boolean isTheSameRequester(RequestAndRelatedRecords it, Request that) {
     return Objects.equals(it.getUserId(), that.getUserId());
   }
 
+  private static Result<RequestAndRelatedRecords> requestLevelTitleAndTlrEnabledErrorMapper(
+    RequestAndRelatedRecords requestAndRelatedRecords) {
+
+    String message = "This requester already has an open request for this instance";
```

Comment:
please inline the variable in calling failedValidation method

Reply:
done